tesseract/tesseract_planning/tesseract_process_planners/include/tesseract_process_planning/process_planner.h

```diff
@@ -0,0 +1,76 @@
+#ifndef TESSERACT_PLANNING_PROCESS_PLANNER_H
+#define TESSERACT_PLANNING_PROCESS_PLANNER_H
+
+#include <tesseract_process_planners/process_definition.h>
+
+namespace tesseract_process_planners
+{
+
+/**
+ * @brief Is a results structure to hold the results generated based on the process segment definition
+ *
+ * An empty trajectory indicates that this process step is to be omitted.
+ *
+ * TODO: Should we store the planner response for each?
+ *
+*/
+struct ProcessSegmentPlan
+{
+  tesseract_common::TrajArray approach;  /**< The generated approach trajectory generated based on the Process Segment Definition */
+  tesseract_common::TrajArray process;   /**< The generated process trajectory generated based on the Process Segment Definition */
+  tesseract_common::TrajArray departure; /**< The generated departure trajectory generated based on the Process Segment Definition */
+  bool valid = true;              /**< If true, a motion plan was sucessfully found for approach, process and departure, otherwise false */
+};
+
+/**
+ * @brief Is a result structure to hold the results generated based on the process definition.
+ *
+ * Note: An empty trajectory for any process step indicates that it is to be omitted.
+ *
+*/
+struct ProcessPlan
+{
+  tesseract_common::TrajArray from_home;                          /**< A trajectory from the home position to the firts point in the first segment */
```

Comment:
We discussed renaming this didn't we? Something like "from_start"

Reply:
That is correct. Lets rename it.

Reply:
done